lib/rust/ensogl/core/src/system/gpu/context/native.rs

```diff
@@ -0,0 +1,116 @@
+//! Extensions for the native [`WebGl2RenderingContext`] implementation.
+
+use crate::prelude::*;
+
+use crate::system::gpu::context::extension::Extensions;
+use crate::system::gpu::shader::Shader;
+
+use web_sys::WebGl2RenderingContext;
+use web_sys::WebGlProgram;
+
+
+#[allow(missing_docs)]
+pub mod traits {
+    pub use super::BlockingCheckStatus;
+    pub use super::BlockingGetErrorLog;
+}
+
+
+// =============================
+// === ContextWithExtensions ===
+// =============================
+
+/// Native WebGL context and supported [`Extensions`].
+#[derive(Clone, Debug, Deref)]
+#[allow(missing_docs)]
+pub struct ContextWithExtensions {
+    #[deref]
+    native:         WebGl2RenderingContext,
+    pub extensions: Extensions,
+}
+
+impl ContextWithExtensions {
+    /// Constructor.
+    pub fn from_native(native: WebGl2RenderingContext) -> Self {
+        let extensions = Extensions::init(&native);
+        Self { native, extensions }
+    }
+}
+
+
+
+// ===========================
+// === BlockingCheckStatus ===
+// ===========================
+
+/// Status check for the given WebGL entity. For shaders, it checks the compilation status, for
+/// shader programs, the linking status. This operation is blocking and should be used only when
+/// necessary. To learn more, see:
+/// [https://developer.mozilla.org/en-US/docs/Web/API/WebGL_API/WebGL_best_practices#compile_shaders_and_link_programs_in_parallel]
+#[allow(missing_docs)]
+pub trait BlockingCheckStatus<T> {
+    fn blocking_check_status(&self, target: &T) -> Option<String>;
```

Comment:
It might be good to explain a bit more about the expected return types. `String` seems surprising at first. 

Same for the other return types in the trait and functions below.

Reply:
I renamed it to `blocking_get_status_logs`.